### ISSUE TITLE
no-catch-shadow false-positive when setting shelljs env to false

### ISSUE THREAD

**Issue by EvgenyOrekhov** [eslint/eslint#9207]
**Tell us about your environment**

* **ESLint Version:** 4.6.0
* **Node Version:** 8.4.0
* **npm Version:** 5.3.0

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
    "env": {
        "shelljs": false
    },
    "rules": {
        "no-catch-shadow": "error"
    }
}
```

</details>

**What did you do? Please include the actual source code causing the issue.**

```js
try {} catch (error) {}
```

**What did you expect to happen?**
No warnings.

**What actually happened? Please include the actual, raw output from ESLint.**
```text
  1:8  error  Value of 'error' may be overwritten in IE 8 and earlier  no-catch-shadow

✖ 1 problem (1 error, 0 warnings)
```

If I remove `"shelljs": false` from the config, the problem goes away.
This bug was not present in ESLint 4.3.0.

**Comment by platinumazure** [comment]
Possibly related to #9202.

@not-an-aardvark If this is related to #9202, maybe we should add a test to `no-catch-shadow` as well?

**Comment by not-an-aardvark** [comment]
Closing as a duplicate of https://github.com/eslint/eslint/issues/9202.

> @not-an-aardvark If this is related to #9202, maybe we should add a test to no-catch-shadow as well?

I suppose so. That said, this affects all rules that use globals, and it's probably not necessary to add a test to each one individually.
